File: app/src/main/java/com/example/layout/ui/slideshow/SlideshowFragment.java
package com.example.layout.ui.slideshow;

import android.arch.lifecycle.ViewModelProvider;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.layout.R;

public class SlideshowFragment extends Fragment {
    private SlideshowViewModel mViewModel;

    public static com.example.layout.ui.slideshow.SlideshowFragment newInstance() {
        return new com.example.layout.ui.slideshow.SlideshowFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewModel = new ViewModelProvider(this, new ViewModelProvider.NewInstanceFactory()).get(SlideshowViewModel.class);
        // TODO: Use the ViewModel
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_slideshow, container, false);
    }
}

File: app/src/main/java/com/example/layout/ui/slideshow/SlideshowFragment.kt
package com.example.layout.ui.slideshow

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProvider.NewInstanceFactory
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.layout.R

class SlideshowFragment : Fragment() {
    private var mViewModel: SlideshowViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mViewModel = ViewModelProvider(this, NewInstanceFactory()).get(
            SlideshowViewModel::class.java
        )
        // TODO: Use the ViewModel
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_slideshow, container, false)
    }

    companion object {
        fun newInstance(): SlideshowFragment {
            return SlideshowFragment()
        }
    }
}